integration/kotlinx-coroutines-guava/src/ListenableFuture.kt

```diff
@@ -436,7 +434,7 @@ private class JobListenableFuture<T>(private val jobToCancel: Job): ListenableFu
     }
 
     /** See [get()]. */
-    private fun getInternal(result: Any): T = if (result is Cancelled) {
+    private fun getInternal(result: Any?): T = if (result is Cancelled) {
         throw CancellationException().initCause(result.exception)
     } else {
         // We know that `auxFuture` can contain either `T` or `Cancelled`.
```

Comment:
Can't the result here be `null` now? In which case, `getInternal` should also return a nullable value.

Reply:
`auxFuture.set` is called in 2 places:
- [`complete`](https://github.com/Kotlin/kotlinx.coroutines/blob/9c3f59fb2a626b96fc5519c3e6c6c55c4d46fdd3/integration/kotlinx-coroutines-guava/src/ListenableFuture.kt#L373), where it's called with a `T`
- [`completeExceptionallyOrCancel`](https://github.com/Kotlin/kotlinx.coroutines/blob/9c3f59fb2a626b96fc5519c3e6c6c55c4d46fdd3/integration/kotlinx-coroutines-guava/src/ListenableFuture.kt#L386), where it's called with a `Cancelled`

(That appears to match [this comment](https://github.com/Kotlin/kotlinx.coroutines/blob/9c3f59fb2a626b96fc5519c3e6c6c55c4d46fdd3/integration/kotlinx-coroutines-guava/src/ListenableFuture.kt#L352).)

It's true that `T` might be a nullable type in some cases, so the result might be `null`. But that's OK: The result can be `null` only if it came through `complete(result: T)`. And _that_ can happen only in the case that `T` is a nullable type.

Put another way, `complete` and `getInternal` combine to implement a system that accepts a `T` and produces a `T`. It just happens to store the `T` as an `Any?` in the meantime because it sometimes wants to store a `Cancelled` instead.

More broadly: This PR shouldn't change any runtime behavior. `result: T` could always be a null value. But once Guava is fully annotated for nullness (maybe in Guava 32.0?), the existing `auxFuture.set(result)` call will stop working: At that point, Kotlin will not only know that `result: T` could be a null value but _also_ know that `auxFuture.set` should not be called with a null value because its parameter type is `Any`. This PR changes `auxFuture` so that `auxFuture.set` accepts an `Any?` and the code continues to compile. But it doesn't change the values that were already possible.